OpenSSL-Strategy.md

```diff
@@ -0,0 +1,361 @@
+
+There are different configurations in use by distributors for linking Node.js to
+OpenSSL. The most notable distributors and the configurations used are:
+
+* **nodejs.org**: Binaries produced by the Node.js Build infrastructure across
+  all supported architectures use the default configuration for OpenSSL and
+  therefore compile and statically link in the OpenSSL source as it ships with
+  Node.js. It should be noted that nodejs.org is the source of binaries for
+  **nvm** which is in heavy client use and is also used by Travis-CI to fetch
+  and install Node.js.
+* **Linux Distributions**: Binaries shipped with Debian, Ubuntu, CentOS, RHEL,
+  Fedora and the majority of other Linux distributions follow a standard policy
+  of separating dependencies and dynamically linking wherever possible.  The
+  Node.js packages on these systems are linked via the packaging dependency
+  mechanisms to OpenSSL 1.0.2 packages and the `node` binaries that ship by
+  default on these platforms use the shared OpenSSL 1.0.2 library installed by
+  those packages.  Therefore, the floating patches do not apply and it is
+  possible that a different version of OpenSSL 1.0.2 is in use by Node.js than
+  the version that was shipped in the source tree.
+* **NodeSource Builds**: Binaries
+  [distributed](https://github.com/nodesource/distributions) by NodeSource are
+  heavily used, and do not follow the Linux distribution policyof dependency
+  seperation. NodeSource uses the Node.js defaults for `configure` and compile
+  and statically link in the OpenSSL source shipped bundled with Node.js.
+* **Homebrew**: The
+  [formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/node.rb)
+  for Node.js as distributed on macOS by the popular `brew` command, by default,
+  will compile using the default `configure` configuration. It is possible to
+  override this using a `with-openssl` option which will compile against the
+  version of OpenSSL that was most recently installed with `brew` but this is
+  not believed to be in common use.
+
+
+## Node.js version-specific strategy
+
+### Node.js version 4.x (EOL 2018-04-30)
+
+No longer maintained.  Not discussed further.
+
+### Node.js versions 6.x (EOL 2019-04-30) and 8.x (EOL 2019-12-31)
+
+Node.js 6.x and 8.x have the same OpenSSL versions and policies.
+
+Node.js 8.x is unusual because it's end-of-life would normally be April, 2020,
+but it was moved earlier to conincide with the end-of-life of OpenSSL 1.0.2.
+
+* OpenSSL version: 1.0.2 (kept up-to-date with latest)
+* Allowed shared OpenSSL version: 1.0.2x
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. No other
+  TLS versions are supported by this version of OpenSSL or Node.js.
+* FIPS: supports being configured to build against OpenSSL FIPS 2.0. FIPS
+  configuration is not default, and OpenSSL FIPS 2.0 is not included in
+  Node.js.
+
+XXX Do we support FIPS builds, or do we treat like other shared builds, best
```

Comment:
I think we support FIPS more than "normal" shared library builds, but I'm not certain.

Reply:
We treated failures in the FIPs builds the same as any other failure, should be blocking.